moyo/src/lib.rs

```diff
@@ -241,6 +243,7 @@ impl MoyoDataset {
             // Space-group type
             number: space_group.number,
             hall_number: space_group.hall_number,
+            hm_symbol: hall_symbol.hm_short.to_string(),
```

Comment:
The current `MoyoDataset` takes one extreme by removing every redundant entry (for example, hm_short is not included because it can be derived from `number`). The other extreme would be to include all secondary entries. 
I was initially unsure how to balance the two extremes and choose secondary information (https://github.com/spglib/moyo/issues/77#issuecomment-2684049654). But on second thought, I think it is reasonable to include `hm_short`. It is frequently needed for quick access, as you pointed out, and [spglib dataset](https://spglib.readthedocs.io/en/stable/dataset.html) offers hall_symbol for convenience